doghouse/appengine/github.go

```diff
@@ -0,0 +1,236 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"github.com/haya14busa/reviewdog/doghouse/server"
+	"github.com/haya14busa/reviewdog/doghouse/server/cookieman"
+	"golang.org/x/oauth2"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/log"
+	"google.golang.org/appengine/urlfetch"
+)
+
+type GitHubHandler struct {
+	clientID     string
+	clientSecret string
+
+	tokenStore     *cookieman.CookieStore
+	redirURLStore  *cookieman.CookieStore // Redirect URL after login.
+	authStateStore *cookieman.CookieStore
+
+	privateKey    []byte
+	integrationID int
+}
+
+func NewGitHubHandler(clientID, clientSecret string, c *cookieman.CookieMan, privateKey []byte, integrationID int) *GitHubHandler {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported function NewGitHubHandler should have comment or be unexported